### PR TITLE
feat(buildpacks): script to checkout and build

### PR DESCRIPTION
**Pull request by mmoreiradj**
## What does this PR do ?

This PR adds a [post-receive hook](https://git-scm.com/docs/githooks#post-update) to check out the project, read the configuration and build it.

### How should this be manually tested ?

Set up a git server using the git-daemon :

```bash
cd /tmp

# init a repo
git init --bare myrepo

# add the hook to the repo
cp /path/to/hooks/post-receive /tmp/myrepo/hooks/post-receive

# give appropriate rights
chmod +x /tmp/myrepo/hooks/post-receive

# start the git daemon
git daemon --verbose --reuseaddr --base-path=/tmp --export-all --informative-errors --enable=receive-pack /tmp

# go to your favorite project
git clone git@github.com:paastech-cloud/website.git
cd website

# add this file for configuration :
touch paastech.yaml

# content
buildpacks:
  path: "."
  builder: "paketobuildpacks/builder:full"
  buildpack: "paketo-buildpacks/web-servers"
  env:
    - "BP_WEB_SERVER=nginx"
    - "BP_WEB_SERVER_ROOT=dist"

# add the remote
git remote add paastech git://127.0.0.1/myrepo

# add files, commit them and push ON MAIN
git push paastech main

# the file should be build and the image name should be "myrepo" which is the name of the directory in which the repo is stored
docker run myrepo:latest
```

### CODE REVIEW

File: hooks/post-receive [hunk 1]
```diff
@@ -1,30 +1,129 @@
...
+# checkout the code to /tmp/app-timestamp
+git --work-tree="$target_path" --git-dir=. checkout -f "$branch"
+
+echo -e "${BLUE}===> READING CONFIGURATION FILE${NC}"
+
+config_path="$target_path"
+
+if [ -f "$config_path/paastech.yaml" ]; then
+    config_path+="/paastech.yaml"
+elif [ -f "$config_path/paastech.yml" ]; then
```
Comment (Hunh0w) [inline hooks/post-receive:60]: why?

Comment (mmoreiradj) [reply hooks/post-receive:60]: it's merged, too late + "why" does not give me much information about what you need.
This piece of code checks is the configuration file paastech.yaml/paastech.yml is present.

Comment (Hunh0w) [reply hooks/post-receive:60]: just read the 3 lines above the "why"

Comment (Hunh0w) [reply hooks/post-receive:60]: there are the same condition twice

Comment (mmoreiradj) [reply hooks/post-receive:60]: Right back at you

**Pull request merged**